daemon/main.go

```diff
@@ -146,6 +147,15 @@ func initConfig() {
 }
 
 func initEnv() {
+	socketDir := path.Dir(socketPath)
+	if err := os.MkdirAll(socketDir, 0700); err != nil {
```

Comment:
Looking further into this. In most cases, the directory is created by the daemon but I notice the mode is set to `755` which we definitely don't want. I will fix this up separately.